src/bounce_solution/action_calculation.cpp

```diff
@@ -0,0 +1,1990 @@
+                       "----------------\tPath deformation\t----------------");
+  for (int it_maxpath = 0; it_maxpath < MaxSinglePathDeformations; it_maxpath++)
+  {
+    NoBestPathCounter++;
+    reductor = reductorCalculator(MaximumGradient) / stepsize;
+
+    oldMaximumRelativeError = MaximumRelativeError;
+
+    SinglePathDeformation(stepsize,
+                          reductor,
+                          l,
+                          rho_l_spl,
+                          l_fornextpath,
+                          best_path,
+                          next_path,
+                          MaximumGradient,
+                          MaximumForce,
+                          MaximumRelativeError,
+                          Maximum_dldrho,
+                          PerpendicularGradient,
+                          inverseK,
+                          forces);
+
+    if (MaximumRelativeError != oldMaximumRelativeError) NoBestPathCounter = 0;
+
+    if (MaximumRelativeError < SatisfactoryError)
+    {
+      BSMPT::Logger::Write(BSMPT::LoggingLevel::BounceDetailed,
+                           "Enough convergence!\t" +
+                               std::to_string(MaximumRelativeError));
+
+      PathDeformationConvergedWithout1D = true;
+      break;
+    }
+    if (MaximumRelativeError > 5) break; // Things went very wrong
+    if (NoBestPathCounter > 20)
+      break; // Algorithm does not seem to find new best paths
+  }
+
+  // Reshift all point to their correct locations
+  if (best_path.size() == 0)
+  {
+    convPathDeformation = -2;
+    action              = path_deformation_crashed;
+    BSMPT::Logger::Write(BSMPT::LoggingLevel::BounceDetailed,
+                         "Path deformation exploded");
+    return;
+  }
+  else if (MaximumRelativeError > SatisfactoryError)
+  {
+    BSMPT::Logger::Write(
+        BSMPT::LoggingLevel::BounceDetailed,
+        "Maximum iterations reached without error increasing. Integrate "
+        "again!\n");
+  }
+
+  for (int it_path = 0; it_path <= NumberPathKnots; it_path++)
+  {
+    best_path[it_path] = FalseVacuum + best_path[it_path];
+  }
+  // Change the class path into the new path
+  setPath(best_path);
+
+  return;
+}
+
+unsigned BounceActionInt::nChoosek(unsigned n, unsigned k)
+{
+  // Auxiliary function to calculate n choose k combination
+  if (k > n) return 0;
+  if (k * 2 > n) k = n - k;
+  if (k == 0) return 1;
+
+  int result = n;
+  for (std::size_t i = 2; i <= k; ++i)
+  {
+    result *= (n - i + 1);
+    result /= i;
+  }
+  return result;
+}
+
+void BounceActionInt::ClearFolder(const char *path_in)
+{
+  std::stringstream ss;
+  ss << path_in;
+
+  /*
+  if (std::experimental::filesystem::exists(path_in))
+  {
+    // Remove everything
+    for (const auto &entry :
+         std::experimental::filesystem::directory_iterator(path_in))
+      std::experimental::filesystem::remove_all(entry.path());
+  }
+  std::experimental::filesystem::create_directory(path_in);
+  */
```

Comment:
What about this code?

Reply:
This was used to empty the folder of the path iterations output. Removed since it is no longer supported.